src/app/sagas/Submission.ts

```diff
@@ -12,10 +12,8 @@ export const getUserLatestCode = (state: RootState) => state.code.code;
 
 export function* lockCode(action: ActionType<typeof SubmissionActions.lockCode>) {
   try {
-    yield put(CodeActions.save());
-    // @ts-ignore
-    // const res = yield call(SubmissionFetch.lockCode);
-    // yield put(NotificationActions.success('Code Locked'));
+    yield call(SubmissionFetch.lockCode);
+    yield put(NotificationActions.success('Code Locked'));
```

Comment:
@ElzaCS @ram2091999 @bestinbthomas Is lockcode something you sill want?
Doesn't it clutter the submitbar?

Reply:
Yes. The code used for matches with other players that affect their rating, should be saved separately. If a user is on the leaderboard, they should be ready to be challenged by another player. In such a situation, if a user is in the middle of improvising their code, the latest submitted code would be used for the match.